tests/testthat/test-qenv_extract.R

```diff
@@ -0,0 +1,68 @@
+
+testthat::test_that("`[.` returns empty qenv for names not in qenv", {
+  data <- within(qenv(), {
+    x <- 1
+    a <- 2
+  })
+  testthat::expect_warning(
+    testthat::expect_equal(data["y"], qenv()),
+    "None of 'names' elements exist in 'qenv'. Returning empty 'qenv'."
+  )
+})
+
+testthat::test_that("`[.` returns limited qenv for some names not in qenv", {
+  data <- within(qenv(), {
+    x <- 1
+    a <- 2
+  })
+  testthat::expect_warning(
+    testthat::expect_equal(data[c("y", "a")], data["a"]),
+    "Some elements of 'names' do not exist in 'qenv'. Skipping those: y."
+  )
+})
+
+testthat::test_that("`[.` extracts proper objects", {
+  q <- qenv()
+  code <- c("x<-1", "a<-1;b<-2")
+  q <- eval_code(q, code)
+  object_names <- c("x", "a")
+  qs <- q[object_names]
+  testthat::expect_true(all(ls(get_env(qs)) %in% object_names))
+})
+
+testthat::test_that("`[.` extract proper code", {
```

Comment:
proper code is not expectation - please be more specific

Reply:
084382a3b6021705c6e1ab968ae4fcc26074d426